test/integration/deployitems/aborting_timeout.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
```

Comment:
I think this test does not need a real k8s cluster.
We could write it as unit test and the envtest lib

Reply:
I wrote unit tests for all timeouts and refactored the integration tests into one single (slightly bigger) test that verifies the timing and cooperation between the deploy item controller and the deployer.